### PR TITLE
feat: add text and emphasis transformers - #397

### LINKED ISSUE #397: CiceroMark<->OOXML transformers
# Feature Request 🛍️

Incorporating the CiceroMark->OOXML transformer and improving the currently implemented OOXML->CiceroMark transformer.

## Use Case

It will allow the conversion of Docx files into CiceroMark JSON and vice versa. In addition, using this `Docx` files can be converted into different other formats like `PDF` or `HTML` as well.

## Possible Solution

The transformer for OOXML->CiceroMark is already implemented. It needs to be updated with some entities to allow full transformation.

The transformer for CiceroMark->OOXML is created in `cicero-word-add-in` branch. It needs to be transferred/transported here with some changes to allow the transformation.

## Detailed Description

Currently, the CiceroMark->OOXML transformer supports the following conversions:
| CiceroMark Entity  | OOXML Entity |
| --------  | -------- |
| Text      | `<w:t>` |
| Paragraph     | `<w:p>Content</w:p>` |
| Linebreak      | `<w:p/>` |
| Softbreak      | `<w:r><w:sym></w:r>` |1
| Emph     | `<w:i>` |
| Variable      | `<w:sdt>` |
| List Block/ List     | `<w:numPr><w:num w:val={ordered/unordered}/></w:numPr>` |
| List Item(Text)     | `<w:t/>` |
| List Item(Variable)     | `<w:sdt/>` |

Conversions which are left:

- `Formula / Ergo expressions`
- `Clause`
- `Conditional`
- `Optional`
- `Strong`
- `BlockQuote`
- `CodeBlock`
- `Code`
- `ThematicBreak`
- `Link`
- `Image`

In the left conversions, we need to decide which ones need major importance/priority and which can be given a lower priority. Furthermore, we also need to think about whether all these will be present in the contract (IMO, Code and CodeBlock generally won't occur in the contract)

### PR DESCRIPTION
**Pull request by K-Kumar-01**
Signed-off-by: k-kumar-01 <kushalkumargupta4@gmail.com>

Reference #397

### Changes

- Rules and helpers for the conversion
- Complete roundtripping for text and emphasis(CiceroMark->OOXML->CiceroMark)
- Test for the above

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

**Comment by K-Kumar-01** [comment]
@algomaster99 Requesting your review.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js [hunk 1]
```diff
@@ -14,23 +14,73 @@
...
-        list: 'org.accordproject.commonmark.List',
-        listBlock: 'org.accordproject.ciceromark.ListBlock',
-        paragraph: 'org.accordproject.commonmark.Paragraph',
-        softbreak: 'org.accordproject.commonmark.Softbreak',
-        text: 'org.accordproject.commonmark.Text',
-        variable: 'org.accordproject.ciceromark.Variable',
-        emphasize: 'org.accordproject.commonmark.Emph',
-    };
+    /**
+     * Gets the class of a given CiceroMark Node.
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:41]: Node should start with a lowercase `n`.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js [hunk 2]
```diff
@@ -14,23 +14,73 @@
...
+     *
+     * @param {object} node    Description of node type
+     * @param {object} counter Counter for different variables based on node name
+     * @param {object} parent  Parent object for a node
+     * @returns {string} OOXML for the given node
+     */
+    getNodes(node, counter, parent = null) {
+        if (this.getClass(node) === definedNodes.text) {
+            if (parent !== null && parent.class === definedNodes.emphasize) {
+                return TEXT_RULE(node.text, true);
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:61]: I would suggest you to create different rules for text and emphasized text. The reason is readability and let me elaborate over this.

What is more readable and quickly understandable?
```js
if (parent !== null &amp;&amp; parent.class === definedNodes.emphasize) {
  return TEXT_RULE(node.text, true);
} else {
  return TEXT_RULE(node.text);
}
```

```js
if (parent !== null &amp;&amp; parent.class === definedNodes.emphasize) {
  return EMPHASIS_RULE(node.text);
} else {
  return TEXT_RULE(node.text);
}
```

I really hope your answer is latter. The point of readability is not just making it easy to read but to understand as welll and that's why naming is important.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js [hunk 3]
```diff
@@ -40,8 +90,12 @@ class CiceroMarkToOOXMLTransfomer {
...
      * @returns {string} Converted OOXML string i.e. CicecoMark->OOXML
      */
-    toOOXML(ciceromark, counter, ooxml) {
-        let globalOOXML = ooxml;
+    toOOXML(ciceromark, counter, ooxml = '') {
+        globalOOXML = ooxml;
+        ciceromark.nodes.forEach(node => {
+            this.getNodes(node, counter);
+        });
+        globalOOXML = wrapOOXML(globalOOXML);
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:98]: A better name would be `wrapAroundDefaultDocxTags`.

File: packages/markdown-docx/test/data/ciceroMark/ciceroMark.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (algomaster99) [inline packages/markdown-docx/test/data/ciceroMark/ciceroMark.js:1]: This is just test data. You don't need a license here.

Comment (algomaster99) [reply packages/markdown-docx/test/data/ciceroMark/ciceroMark.js:1]: Convert this file so that it only contains JSON data. Using JS is just making it verbose.

File: packages/markdown-docx/src/rules.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (algomaster99) [inline packages/markdown-docx/src/rules.js:1]: You should rename this file to `toOOXMLRules.js`. Since this package has two transformers - `OOXML -> CiceroMark` and `CiceroMark -> OOXML`, it will help remove the ambiguity.

File: packages/markdown-docx/test/data/ciceroMark/ciceroMark.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const paraAndEmphasisJSON = {
+    '$class':'org.accordproject.commonmark.Document',
+    'xmlns':'http://commonmark.org/xml/1.0',
+    'nodes':[
```
Comment (algomaster99) [inline packages/markdown-docx/test/data/ciceroMark/ciceroMark.js:20]: Could you create a `.cta` archive for this? I can then check with the word add in too.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Once you create the `.cta` archive, I can give more reviews on the logic for emphasis vistor. For now, see comments.

- Download VSCode extension for JS formatting so that your white spaces are consistent. Make sure you set the rules which are consistent with the codebase.
- Push your branch to this repository itself (not your fork). Makes it easier for me to fetch and test.

File: packages/markdown-docx/test/data/ciceroMark/ciceroMark.js (hunk 6 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/test/data/ciceroMark/ciceroMark.js:20]: @algomaster99 
Actually, I generated the JSON using the template mark dingus and not `.cta`. However, I will create a `.cta` file for it and send

File: packages/markdown-docx/test/data/ciceroMark/ciceroMark.js (hunk 4 above)
Comment (algomaster99) [reply packages/markdown-docx/test/data/ciceroMark/ciceroMark.js:1]: Yeah, that would be better. Separate files for each test.

**Comment by K-Kumar-01** [comment]
> * Push your branch to this repository itself (not your fork). Makes it easier for me to fetch and test.

Can I do this from next PR?

**Comment by algomaster99** [comment]
@K-Kumar-01 yes. I forgot to add "do this from the next PR".

**Comment by K-Kumar-01** [comment]
@algomaster99 
I have updated the code.
Also here is the cta.
[test-and-emphasis.zip](https://github.com/accordproject/markdown-transform/files/6634098/test-and-emphasis.zip)
Let me know if there are any more changes required.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js [hunk 7]
```diff
@@ -14,23 +14,73 @@
...
+    list: 'org.accordproject.commonmark.List',
+    listBlock: 'org.accordproject.ciceromark.ListBlock',
+    paragraph: 'org.accordproject.commonmark.Paragraph',
+    softbreak: 'org.accordproject.commonmark.Softbreak',
+    text: 'org.accordproject.commonmark.Text',
+    variable: 'org.accordproject.ciceromark.Variable',
+    emphasize: 'org.accordproject.commonmark.Emph',
+};
+
+let globalOOXML = '';
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:33]: Why did you move this outside the class? This should be an instance variable inside the constructor.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+'use strict';
+
+const { sanitizeHtmlChars } = require('./CiceroMarkToOOXMLHelpers');
+
+/**
+ * Inserts text.
+ *
+ * @param {string}  value     Text to be rendered
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:22]: You are not rendering anything. This was true for the add-in but not markdown transform. Better wording would be `enclosing value of the OOXML tag`

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {string} OOXML for the text
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:23]: OOXML **tag** for the text

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+ */
+const TEXT_RULE = (value) => {
+    return `<w:r><w:t xml:space="preserve">${sanitizeHtmlChars(value)}</w:t></w:r>`;
+};
+
+/**
+ * Inserts emphsaised text.
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:30]: A typographical error here. "emphasised"

File: packages/markdown-docx/src/OoxmlTransformer.js [hunk 11]
```diff
@@ -151,6 +151,17 @@ class OoxmlTransformer {
...
+                let emphFound = element.elements[0].elements.some(subElement=>{
+                    return subElement.name==='w:i' && subElement.attributes['w:val'] === 'true';
+                });
+                if(emphFound){
+                    return {
+                        $class: `${NS_PREFIX_CommonMarkModel}Emph`,
+                        nodes:[...this.deserializeElements(element.elements)]
+                    };
+                }
+            }
```
Comment (algomaster99) [inline packages/markdown-docx/src/OoxmlTransformer.js:164]: I will see to this later in the morning.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Dropped a few comments.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js (hunk 7 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:33]: @algomaster99 
[Here](https://github.com/accordproject/markdown-transform/pull/400), you suggested it move out of the class. EsLint was unable to parse this as a class variable.

Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:33]: @algomaster99 Made the other suggested changes.

Comment (algomaster99) [reply packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:33]: Class variables and instance variables are different. Instance variables are specific to the object of the class. They are accessed using the `this` keyword. [Example](https://github.com/accordproject/markdown-transform/blob/master/packages/markdown-cicero/lib/CiceroMarkTransformer.js#L47).

Class variables or static variables are common for all objects of the class. They don't even require you to instantiate a class to access them.

Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:33]: I think I misunderstood this one :sweat_smile: . I will revert it and push.

Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js]: @algomaster99 Done

File: packages/markdown-docx/src/OoxmlTransformer.js [hunk 12]
```diff
@@ -151,6 +151,17 @@ class OoxmlTransformer {
                 $class: `${NS_PREFIX_CommonMarkModel}Softbreak`,
             };
         case 'w:r':
+            if(element.elements[0].name==='w:rPr'){
+                let emphFound = element.elements[0].elements.some(subElement=>{
+                    return subElement.name==='w:i' && subElement.attributes['w:val'] === 'true';
```
Comment (algomaster99) [inline packages/markdown-docx/src/OoxmlTransformer.js:156]: > subElement.attributes['w:val'] === 'true';

This is not necessary. Italics is valid without it too.

File: packages/markdown-docx/src/OoxmlTransformer.js (hunk 11 above)
Comment (algomaster99) [reply packages/markdown-docx/src/OoxmlTransformer.js:164]: Improve the spacing by using a good formatter such as prettier. You can integrate its extension in VSCode. Make sure the rules your configure are _almost_ consistent with the entire codebase.

File: packages/markdown-docx/src/OoxmlTransformer.js [hunk 13]
```diff
@@ -151,6 +151,17 @@ class OoxmlTransformer {
...
+                });
+                if(emphFound){
```
Comment (algomaster99) [inline packages/markdown-docx/src/OoxmlTransformer.js:158]: `emphasisedTextFound`

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Final reviews then we are good to merge.

File: packages/markdown-docx/src/OoxmlTransformer.js (hunk 12 above)
Comment (algomaster99) [reply packages/markdown-docx/src/OoxmlTransformer.js:156]: In fact, we would need to delete this attribute from the `CIceroMark -> OOXML` transformer too.

File: packages/markdown-docx/src/OoxmlTransformer.js (hunk 13 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/OoxmlTransformer.js:158]: @algomaster99 
I used the default JS and Ts formatters above.
Also, for prettier, there was no config file I didn't use it. I will use the default config for prettier that match the codebase.

Comment (algomaster99) [reply packages/markdown-docx/src/OoxmlTransformer.js:158]: Yes, use prettier. It should take care of everything. Just make sure it doesn't change very unrelated things in codebase. For example, removes all the `;`. This is why I am asking you to write a config file. You can check the docs on how to write one. It's very straightforward.

Comment (K-Kumar-01) [reply packages/markdown-docx/src/OoxmlTransformer.js]: @algomaster99 
I have formatted using prettier. Let me know if there are any more changes required.

Comment (algomaster99) [reply packages/markdown-docx/src/OoxmlTransformer.js]: Sure. Go ahead. Do that in a separate PR though. And then we can merge the changes with this PR.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+/**
+ * Inserts emphasised text.
+ *
+ * @param {string}  value  Enclosing value of the OOXML tag
+ * @returns {string} OOXML tag for the emphasised text
+ */
+const EMPHASIS_RULE = (value) => {
+    return `<w:r><w:rPr><w:i w:val="true" /></w:rPr><w:t>${sanitizeHtmlChars(value)}</w:t></w:r>`;
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:36]: `w:val="true"` is not needed like I reasoned before.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+const { sanitizeHtmlChars } = require('./CiceroMarkToOOXMLHelpers');
+
+/**
+ * Inserts text.
+ *
+ * @param {string}  value Enclosing value of the OOXML tag
+ * @returns {string} OOXML tag for the text
+ */
+const TEXT_RULE = (value) => {
+    return `<w:r><w:t xml:space="preserve">${sanitizeHtmlChars(value)}</w:t></w:r>`;
```
[the remark this answers is not in the document]
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: @algomaster99 These tags are not used for conversion. So any space between them will act as '    <value>' instead of '<value>'. However, in `wrapAroundDefaultDocxTags` we do not check for such. So no omission of spaces there.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 14 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:36]: @algomaster99 
It was present in the initial converter (cicero-add-in) so I didn't change it. Should I remove this/

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+const OoxmlTransformer = require('./OoxmlTransformer');
+const CiceroMarkToOOXMLTransfomer = require('./CiceroMarkToOOXMLTransformer');
+
+describe('Perform roundtripping between CiceroMark and OOXML', () => {
+
+    it('should parse paragraphs and emphasis nodes.', async () => {
+        const paraAndEmphasisCiceroMark = await fs.readFileSync('test/data/ciceroMark/text-and-emphasis.json', 'utf-8');
+
+        const ciceroMarkTransformer = new CiceroMarkToOOXMLTransfomer();
+        const ooxml = ciceroMarkTransformer.toOOXML(JSON.parse(paraAndEmphasisCiceroMark));
```
[the remark this answers is not in the document]
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js:33]: Yes, they are required. The tests are failing if I do not add `JSON.parse`. 
Screenshot from 2021-06-11 13-56-12

File: packages/markdown-docx/src/OoxmlTransformer.js (hunk 13 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/OoxmlTransformer.js]: @algomaster99 I didn't made any drastic prettier change. Only small area of code needed proper spacing so I don't think it would need a PR.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 14 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js]: @algomaster99 
Done

**Comment by algomaster99** [comment]
Can you answer [this](https://github.com/accordproject/markdown-transform/pull/401#discussion_r649782324) first?

Also, be super clear about what you're testing. In some places, it is `text-and-emphasis` and in some places, it's `paraAndEmphasis`. It would cause confusion to someone who is new to this repository.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (algomaster99) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: > So any space between them will act as ' <value>' instead of '<value>'

Any problems with that? This worked fine in the Word add-in. I think the parser we use, `xml-js`, is also smart enough to ignore such spaces.

[the remark this answers is not in the document]
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: @algomaster99
It included the spaces also in  values of nodes if it is done similar to add-in. So i dedcided to do it in later way. I also posted pn Slack regarding this. I will also reconfirm it.

[the remark this answers is not in the document]
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: @algomaster99 
Screenshot from 2021-06-11 15-01-12
Changed the `EMPHASIS_RULE` and failed test

**Comment by K-Kumar-01** [comment]
@algomaster99 
> Okay, right! Better to just parse it once and comment _why_ you needed to parse it.

The comment I made is: `converts from string to JSON object`.

Should I do a final push? I have updated the names to `textAndEmphasis` to be more clear.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (algomaster99) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: By value, do you mean - `<w:value />` or `<w:t>value</w:t>`?

I read your post on slack and I thought it was causing some problems for you. But I don't think there is any issue as I just tried on my local with spaces and breaks.

If you think the size of the string (in bytes) is an issue, you can add a package to uglify an OOXML string after transformation is done.

**Comment by algomaster99** [comment]
Address [this comment](https://github.com/accordproject/markdown-transform/pull/401#discussion_r649837073) and we would be good to go.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (algomaster99) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: That is because you literally put a space before and after your value. I am just asking to format the rest of the nodes.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 14 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js]: > Address [this comment](https://github.com/accordproject/markdown-transform/pull/401#discussion_r649837073) and we would be good to go.

@algomaster99 
I already made the changes for this.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (algomaster99) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: Let's discuss this in the meeting.

**Comment by K-Kumar-01** [comment]
> Address [this comment](https://github.com/accordproject/markdown-transform/pull/401#discussion_r649837073) and we would be good to go.



> Address [this comment](https://github.com/accordproject/markdown-transform/pull/401#discussion_r649837073) and we would be good to go.

@algomaster99 
The changes for this were already made by me in [this](https://github.com/accordproject/markdown-transform/pull/401/commits/18fb1fdb8e7d7d34bdf3ecd0cbf755f4fde173d8) commit.

Should i push the final changes?

**Comment by algomaster99** [comment]
Your solution to avoid the test failing because of spaces is at the expense of making the code less readable. I explained a better solution above. Read it if you have time otherwise, we will discuss in the meeting.

File: packages/markdown-docx/src/CiceroMarkToOOXMLRules.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXMLRules.js:26]: > By value, do you mean - `<w:value />` or `<w:t>value</w:t>`?

I mean `<w:t>value</w:t>

>I read your post on slack and I thought it was causing some problems for you. But I don't think there is any issue as I just tried on my local with spaces and breaks.

Interesting, as it gave me errors (shown above).

> If you think the size of the string (in bytes) is an issue, you can add a package to uglify an OOXML string after transformation is done.

We can have a discussion about this in the meeting.

File: packages/markdown-docx/src/CiceroMarkToOOXML/rules.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+ * @param {string}  value  Enclosing value of the OOXML tag
+ * @returns {string} OOXML tag for the emphasised text
+ */
+const EMPHASIS_RULE = (value) => {
+    return `<w:r>
+        <w:rPr>
+            <w:i />
+        </w:rPr>
+        <w:t>${sanitizeHtmlChars(value)}</w:t>
+    </w:r>`;
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXML/rules.js:43]: If you're putting efforts to indent them, do them properly.
```js
return `
    <w:r>
        <w:rPr>
            <w:i />
        </w:rPr>
        <w:t>${sanitizeHtmlChars(value)}</w:t>
    </w:r>
`;
```

Comment (algomaster99) [reply packages/markdown-docx/src/CiceroMarkToOOXML/rules.js:43]: Same goes for above.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
I did not notice the formatting in the meeting. Just push this and I will merge.

File: packages/markdown-docx/src/CiceroMarkToOOXML/rules.js (hunk 17 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/CiceroMarkToOOXML/rules.js]: @algomaster99 Done

**Comment by algomaster99** [comment]
You accidentally pushed the prettier configuration too. I advise you to never use `git add .`.

**Comment by K-Kumar-01** [comment]
> You accidentally pushed the prettier configuration too. I advise you to never use `git add .`.

@algomaster99 
How can we remove it?

**Review by algomaster99 — APPROVED** [review]
Great work!
